libclamav/udf.h

```diff
@@ -0,0 +1,527 @@
+
+typedef struct __attribute__((packed)) {
+    DescriptorTag tag;
+    uint32_t volumeDescriptorSequenceNumber;
+
+    regid implementationIdentifier;
+    uint8_t implementationUse[460];
+
+} ImplementationUseVolumeDescriptor;
+
+/* https://www.ecma-international.org/wp-content/uploads/ECMA-167_3rd_edition_june_1997.pdf */
+/* 4/3 */
+typedef struct __attribute__((packed)) {
+    uint32_t logicalBlockNumber;
+
+    uint16_t partitionReferenceNumber;
+} LBAddr;
+
+//https://www.ecma-international.org/wp-content/uploads/ECMA-167_3rd_edition_june_1997.pdf
+//section 4/23
+typedef struct __attribute__((packed)) {
+    uint32_t priorRecordedNumberOfDirectEntries;
+    uint16_t strategyType;
+    uint8_t strategyParameter[2]; /*described as 'bytes' in docs, so don't want to worry about byte order.*/
+    uint16_t maxEntries;
+    uint8_t reserved_must_be_zero;
+
+    uint8_t fileType;
+
+    LBAddr parentICBLocation;
+
+    uint16_t flags;
+} ICBTag;
+
+typedef struct __attribute__((packed)) {
+
+    DescriptorTag tag;
+
+    uint16_t versionNumber;
+
+    uint8_t characteristics;
+
+    uint8_t fileIdentifierLength;
+
+    long_ad icb;
+
+    uint16_t implementationLength;
+
+    uint8_t rest[1];
+
+} FileIdentifierDescriptor;
+
+#define FILE_IDENTIFIER_DESCRIPTOR_SIZE_KNOWN (sizeof(FileIdentifierDescriptor) - 1)
+
+/*Section 14.4.9 of https:... */
+static uint32_t getFileIdentifierDescriptorPaddingLength(const FileIdentifierDescriptor* const fid)
+{
+    uint32_t ret = 0;
+    uint32_t tmp = fid->implementationLength + fid->fileIdentifierLength + 38;
+    ret          = tmp + 3;
+    ret          = ret / 4;
+
+    ret = ret * 4;
+    ret = ret - tmp;
+
+    return ret;
+}
+
+static inline size_t getFileIdentifierDescriptorSize(const FileIdentifierDescriptor* fid)
+{
+
+    return FILE_IDENTIFIER_DESCRIPTOR_SIZE_KNOWN + fid->implementationLength + fid->fileIdentifierLength + getFileIdentifierDescriptorPaddingLength(fid);
+}
+
+typedef struct __attribute__((packed)) {
+    DescriptorTag tag;
+
+    ICBTag icbTag;
+
+    uint32_t uid;
+
+    uint32_t gid;
+
+    uint32_t permissions;
+
+    uint16_t fileLinkCnt;
+
+    uint8_t recordFormat;
+    uint8_t recordDisplayAttributes;
+
+    uint32_t recordLength;
+
+    uint64_t infoLength;
+
+    uint64_t logicalBlocksRecorded;
+
+    timestamp accessDateTime;
+
+    timestamp modificationDateTime;
+
+    timestamp attributeDateTime;
+
+    uint32_t checkpoint;
+
+    long_ad extendedAttrICB;
+
+    regid implementationId;
+
+    uint64_t uniqueId;
+
+    uint32_t extendedAttrLen;
+
+    uint32_t allocationDescLen;
+
+    /* Variable length stuff here, need to handle;
+    */
+    uint8_t rest[1];
+
+} FileEntryDescriptor;
+
+#define FILE_ENTRY_DESCRIPTOR_SIZE_KNOWN (sizeof(FileEntryDescriptor) - 1)
+static inline size_t getFileEntryDescriptorSize(const FileEntryDescriptor* fed)
+{
+    return FILE_ENTRY_DESCRIPTOR_SIZE_KNOWN + fed->extendedAttrLen + fed->allocationDescLen;
+}
+
+typedef struct {
```

Comment:
Seems like this struct isn't used.  Do you want to keep it? 

If so, should it be `__attribute__((packed))`?